### ISSUE TITLE
Unhelpful error message with circular dependency

### ISSUE THREAD

**Issue by DLu** [catkin/catkin_tools#229]
If you have a circular dependency, catkin_make outputs a decent error message. 

    CMake Error at /home/dlu/test_workspace/build/catkin_generated/order_packages.cmake:7 (message):
      Circular dependency in subset of packages:

      docbrown, marty

However, catkin build results in this error:

    Found '2' packages in 0.0 seconds. 
    [build] Runtime: 0.0 seconds 
    Traceback (most recent call last):
      File "/usr/bin/catkin", line 9, in <module>
        load_entry_point('catkin-tools==0.3.0', 'console_scripts', 'catkin')()
      File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 219, in main
        sys.exit(args.main(args) or 0)
      File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/cli.py", line 296, in main
        summarize_build=opts.summarize  # Can be True, False, or None
      File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 601, in     build_isolated_workspace
        packages_to_be_built = topological_order_packages(dict(packages_to_be_built))
      File "/usr/lib/python2.7/dist-packages/catkin_pkg/topological_order.py", line 138, in topological_order_packages
        if package.name in decorators_by_name:
    AttributeError: 'str' object has no attribute 'name'

**Comment by DLu** [comment]
http://gonzo.probablydavid.com/stuff/test_workspace.tar.gz

**Comment by wjwwood** [comment]
Looks like the error is coming up from https://github.com/ros-infrastructure/catkin_pkg/blob/master/src/catkin_pkg/topological_order.py#L117 which is in `catkin_pkg`. It might be that we have to check for circular dependencies before passing to that function or there is a regression in that function. Not sure.

I don't have time right now to dig into it, but I can try to have a look sometime next week.

**Comment by IanTheEngineer** [comment]
Good find, David. Here's a reproduction of the bug, visualized:
![](http://33.media.tumblr.com/07329ebbf15aa37f8dedf296a0f5586b/tumblr_n1xczxDoQe1r5dm0so1_500.gif)

**Comment by jbohren** [comment]
This is fixed in #249.

**Comment by DLu** [comment]
+1

**Comment by kopp** [comment]
I get the following error with catkin 0.4.2 (Ubuntu 14.04) with circular dependencies (a depends on b, b depends on a):

```
[build] Found '2' packages in 0.0 seconds.                                                                                                                                                                                                                                                                                 
Traceback (most recent call last):
  File "/usr/bin/catkin", line 9, in <module>
    load_entry_point('catkin-tools==0.4.2', 'console_scripts', 'catkin')()
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 258, in main
    catkin_main(sysargs)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 253, in catkin_main
    sys.exit(args.main(args) or 0)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/cli.py", line 418, in main
    summarize_build=opts.summarize  # Can be True, False, or None
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 305, in build_isolated_workspace
    packages, context, workspace_packages)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 98, in determine_packages_to_be_built
    workspace_package_names = dict([(pkg.name, (path, pkg)) for path, pkg in ordered_packages])
AttributeError: 'str' object has no attribute 'name'
```

This looks slightly different and as far as I understood the bug above is fixed in 0.4.x -- right?  So I'm not sure whether this is something new...

**Comment by timotheecour** [comment]
reopened here since found same issue: https://github.com/catkin/catkin_tools/issues/430
